byway/src/main/java/com/google/sps/data/Trip.java

```diff
@@ -64,6 +66,15 @@ public Trip(
     this.route = new ArrayList<String>(route);
   }
 
+  /**
+   * Set the interests for the trip as plain text with the Collection passed in.
+   *
+   * @param interests collection of Strings indicating a user interest
+   */
+  public void setInterests(Collection<String> interests) {
+    this.interests = new ArrayList<>(interests);
```

Comment:
No need to make `interests` non-final and allocate a new array, you can just use `interests.clear(); interests.addAll(interests);` One nice thing about this is that if the inner array of `interests` is large enough to hold the new number of interests, it avoids reallocation, and can just copy the values from the input.

Reply:
Fixed